### PR TITLE
feat(mesh): subscribe stories

### PR DESCRIPTION
**Pull request by tyc67**
## Notable Changes

- user subscribed stories page [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=9004-242058&m=dev)

Screenshot 2024-08-21 at 5 20 38 PM
Screenshot 2024-08-21 at 5 21 09 PM
Screenshot 2024-08-21 at 5 21 43 PM

https://github.com/user-attachments/assets/e2459ffb-8bcc-4180-8785-85837db0bd57

https://github.com/user-attachments/assets/0d870a10-8d86-49a8-a373-6ade84925cb4

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207975510192858

### CODE REVIEW

File: packages/mesh/app/actions/subscribe-stories.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    unlockStoriesResponse
+      ?.map((data) => {
+        if (data.unlockStory && data.unlockStory.source) {
+          return {
+            ...data.unlockStory,
+            expireDate: data.expireDate,
+          }
+        }
+      })
+      .filter(Boolean) ?? []
```
Comment (LinHeMa) [inline packages/mesh/app/actions/subscribe-stories.ts:35]: 獲益良多

**Comment by LinHeMa** [comment]
我目前填寫完驗證碼，似乎無法看到points頁面

File: packages/mesh/app/actions/subscribe-stories.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    globalLogFields,
+    'Failed to get unlock stories'
+  )
+  const unlockStoriesResponse = prev
+    ? response?.member?.transaction?.slice(prev) || []
+    : response?.member?.transaction || []
+  const unlockStories =
+    unlockStoriesResponse
+      ?.map((data) => {
```
Comment (erase2004) [inline packages/mesh/app/actions/subscribe-stories.ts]: `?.` 應該多餘的？

File: packages/mesh/app/point/subscribe-stories/layout.tsx
Comment (erase2004) [inline packages/mesh/app/point/subscribe-stories/layout.tsx:1]: 這個基本上和 https://github.com/readr-media/Sachiel/blob/dev/packages/mesh/app/point/sponsorship/layout.tsx 這一樣的，未來可以考慮用 [Route Group](https://nextjs.org/docs/app/building-your-application/routing/route-groups) 之類的把兩者給整併？

**Review by erase2004 — COMMENTED** [review]
GraphQL 抓資料可以藉由 take 和 skip 來達成 pagination 的效果，
skip 設定起始點，take 設定抓取數量

**Comment by tyc67** [comment]
> GraphQL 抓資料可以藉由 take 和 skip 來達成 pagination 的效果， skip 設定起始點，take 設定抓取數量

修改於 [f740144](https://github.com/readr-media/Sachiel/pull/830/commits/f740144a12abe3e494e1bd84f4784974379e811b)

File: packages/mesh/app/actions/subscribe-stories.ts (hunk 2 above)
Comment (tyc67) [reply packages/mesh/app/actions/subscribe-stories.ts]: 修改於 [f740144](https://github.com/readr-media/Sachiel/pull/830/commits/f740144a12abe3e494e1bd84f4784974379e811b)

File: packages/mesh/app/point/subscribe-stories/layout.tsx
Comment (tyc67) [reply packages/mesh/app/point/subscribe-stories/layout.tsx:1]: Route Group 後續和 caesarWHLee 討論導入layout的方式，暫時整理於 [7aa4020](https://github.com/readr-media/Sachiel/pull/830/commits/7aa402013a4f4655a7899559e0ecd2892a35fafe)

**Comment by tyc67** [comment]
> 我目前填寫完驗證碼，似乎無法看到points頁面

已更新 cms data

File: packages/mesh/graphql/query/member.gql [hunk 3]
```diff
@@ -438,25 +434,26 @@ query GetMemberTransactions($memberId: ID!, $take: Int) {
   }
 }
 
-query GetMemberUnlockStories($memberId: ID!, $take: Int) {
+query GetMemberUnlockStories($memberId: ID!, $skip: Int, $take: Int) {
```
Comment (erase2004) [inline packages/mesh/graphql/query/member.gql]: `$skip: Int`  可以調整為 `$skip: Int = 0`，讓它有預設值

**Review by erase2004 — APPROVED** [review]
Accepted 👍 
另外，記得先處理 conflict 再 merge

**Review by LinHeMa — APPROVED** [review]
LGTM

**Pull request merged**